### PR TITLE
Bootified Module Launcher 

### PR DESCRIPTION
**Pull request by mbogoevici**

### CODE REVIEW

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java [hunk 1]
```diff
@@ -68,6 +72,10 @@
 	 * may be null or empty if the local repository is off line.
 	 */
 	public AetherModuleResolver(File localRepository, Map<String, String> remoteRepositories) {
+		if (log.isDebugEnabled()) {
+			log.debug("Local repository: " + localRepository);
+			log.debug("Remote repository: " + remoteRepositories);
```
Comment (markfisher) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java]: technically this should have "Remote repositories: "

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherApplication.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+@SpringBootApplication
+@ComponentScan(basePackageClasses = ModuleLauncherApplication.class)
+public class ModuleLauncherApplication {
+
+	private static final Log log = LogFactory.getLog(ModuleLauncherApplication.class);
+
+	public static void main(String[] args) throws Exception {
+		try {
+			SpringApplication.run(ModuleLauncherApplication.class, args);
+		} catch (Exception e) {
```
Comment (markfisher) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherApplication.java]: should be a new line after the closing curly brace ;)

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java (hunk 1 above)
Comment (mbogoevici) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java]: Some checks are required there too, I think.

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherApplication.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+			log.error("Couldn't launch module: ", e);
```
Comment (markfisher) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherApplication.java]: no module string follows the colon (also maybe "Failed to launch module" would be better).

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java [hunk 4]
```diff
@@ -49,10 +49,7 @@
 public class ModuleLauncher {
```
Comment (markfisher) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java]: JavaDoc for this class should be updated, and most of what's here should move to ModuleLauncherApplication

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java [hunk 5]
```diff
@@ -63,13 +60,14 @@
 
 	private final ModuleResolver moduleResolver;
 
+	@Deprecated
 	public ModuleLauncher() {
-		this(determineLocalRepositoryLocation(),
-				Collections.singletonMap("spring-cloud-stream-modules", determineRemoteRepositoryLocation()));
+		this(new AetherModuleResolver(new File(ModuleLauncherConfiguration.DEFAULT_LOCAL_REPO),
+				Collections.singletonMap("spring-cloud-stream-modules",ModuleLauncherConfiguration.DEFAULT_REMOTE_REPO)));
```
Comment (markfisher) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java]: nitpick: should be a space after the comma

**Comment by mbogoevici** [comment]
Updated JavaDoc

**Comment by markfisher** [comment]
rebased, squashed, and merged as https://github.com/spring-cloud/spring-cloud-stream/commit/31daf48cc296397693eac844e2e53aa2637a84fe

**Pull request closed without merging**